atom/renderer/api/atom_api_spell_check_client.cc

```diff
@@ -148,13 +161,9 @@ bool SpellCheckClient::SpellCheckWord(const base::string16& word_to_check) {
 // This function is a fall-back when the SpellcheckWordIterator class
 // returns a concatenated word which is not in the selected dictionary
 // (e.g. "in'n'out") but each word is valid.
-bool SpellCheckClient::IsValidContraction(const base::string16& contraction) {
-  if (!contraction_iterator_.IsInitialized() &&
-      !contraction_iterator_.Initialize(&character_attributes_, false)) {
-    // We failed to initialize the word iterator, return as spelled correctly.
-    VLOG(1) << "Failed to initialize contraction_iterator_";
-    return true;
-  }
+bool SpellCheckClient::IsValidContraction(const SpellCheckScope& scope,
+                                          const base::string16& contraction) {
+  DCHECK(contraction_iterator_.IsInitialized());
```

Comment:
:+1: to initializing this once in SpellCheckWord() and turning all that checking into a single DCHECK here